library/src/cache/patch_manager.rs

```diff
@@ -195,25 +195,30 @@ impl PatchManager {
             .with_context(|| format!("Failed to delete patch dir {}", &patch_dir.display()))
     }
 
-    /// Attempts to use the last successfully booted patch as the next boot patch. If the last successfully
-    /// booted patch is not bootable or has the same number as the patch we're falling back from, we clear it.
-    fn try_fall_back_to_last_booted_patch(&mut self) {
+    /// Deletes artifacts for the provided bad_patch_number and attempts to set the next_boot_patch to the last
+    /// successfully booted patch. If the last successfully booted patch is not bootable or has the same number
+    /// as the patch we're falling back from, we clear it as well.
+    fn try_fall_back_from_patch(&mut self, bad_patch_number: usize) {
+        let _ = self.delete_patch_artifacts(bad_patch_number);
```

Comment:
It appears you're intentionally swallowing Err here?  Are you sure that's what you want?

Reply:
`delete_patch_artifacts` logs the error – we don't need to do anything with it here. Failing to clear bad patch artifacts shouldn't stop us from marking the patch as bad and verifying that we can roll back (or not)

Reply:
I think you'll want to leave your future self that as a comment. :)

Reply:
Caused me to re-read https://stackoverflow.com/questions/51141672/how-do-i-ignore-an-error-returned-from-a-rust-function-and-proceed-regardless just now. :)

Reply:
> I think you'll want to leave your future self that as a comment. :)

Done